src/main/java/duck/Event.java

```diff
@@ -0,0 +1,62 @@
+package duck;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Random;
+
+/**
+ * Represents an Event task with a start and end time.
+ */
+public class Event extends Task {
+    protected LocalDateTime from;
+    protected LocalDateTime to;
+
+    /**
+     * Constructs an Event task.
+     *
+     * @param isDone Whether the event is completed.
+     * @param description The description of the event.
+     * @param from The start time of the event in "yyyy-MM-dd HHmm" format.
+     * @param to The end time of the event in "yyyy-MM-dd HHmm" format.
+     */
+    public Event(Boolean isDone, String description, String from, String to) {
+        super(isDone, description);
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+        Random r = new Random();
+        this.from = from.equalsIgnoreCase("now") ? LocalDateTime.now()
+                : from.equalsIgnoreCase("later") ? LocalDateTime.now().plusMinutes(r.nextInt(10000))
+                : LocalDateTime.parse(from.trim(), formatter);
+        this.to = to.equalsIgnoreCase("now") ? LocalDateTime.now()
+                : from.equalsIgnoreCase("later") ? LocalDateTime.now().plusMinutes(r.nextInt(10000))
+                : LocalDateTime.parse(to.trim(), formatter);
```

Comment:
Should magic numbers be `static final` variables?

```suggestion
        static final int INTERVAL_BOUND = 10000;
        this.from = from.equalsIgnoreCase("now") ? LocalDateTime.now()
                : from.equalsIgnoreCase("later") ? LocalDateTime.now().plusMinutes(r.nextInt(INTERVAL_BOUND))
                : LocalDateTime.parse(from.trim(), formatter);
        this.to = to.equalsIgnoreCase("now") ? LocalDateTime.now()
                : from.equalsIgnoreCase("later") ? LocalDateTime.now().plusMinutes(r.nextInt(INTERVAL_BOUND))
                : LocalDateTime.parse(to.trim(), formatter);
```